src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php

```diff
@@ -77,9 +77,16 @@ public function __construct(TokenStorageInterface $tokenStorage, UserProviderInt
     public function handle(GetResponseEvent $event)
     {
         $request = $event->getRequest();
-        $username = $request->get($this->usernameParameter) ?: $request->headers->get($this->usernameParameter);
 
-        if (!$username) {
+        // usernames can be falsy
+        $username = $request->get($this->usernameParameter);
+
+        if (null === $username) {
```

Comment:
What about the empty string? Should we exclude it too?

Reply:
Good idea. Thanks for catching that.